File: 2308-divide-array-into-equal-pairs/2308-divide-array-into-equal-pairs.py
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        map1 = dict()

        for num in nums:
            map1[num] = map1.get(num, 0) + 1
        
        for value in map1.values():
            if(value%2!=0):
                return False
        return True
            
        

File: 2308-divide-array-into-equal-pairs/2308-divide-array-into-equal-pairs.cpp
class Solution {
public:
    bool divideArray(vector<int>& nums) {
        unordered_map<int,int> map;
        for(int n:nums){
            map[n]++;
        }
        bool isValid = true;
        for(auto& it: map){
            if(it.second%2!=0){
                isValid = false;
            }
        }
        return isValid;
    }
};